### PR TITLE
adding pdb to test to pass through

### PR DESCRIPTION
**Pull request by jmikedupont2**
##### SUMMARY
Be able to pass through --pdb like pytest

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
testing

##### ADDITIONAL INFORMATION
passthrough `--pdb`

### CODE REVIEW

File: test/lib/ansible_test/_internal/config.py [hunk 1]
```diff
@@ -298,6 +298,7 @@ def __init__(self, args):
 
         self.collect_only = args.collect_only  # type: bool
         self.num_workers = args.num_workers  # type: int
+        self.pdb = args.pdb == 'yes'
```
Comment (mattclay) [inline test/lib/ansible_test/_internal/config.py:301]: What is the purpose of this new config? It isn't used anywhere.

**Comment by jmikedupont2** [comment]
It is used by me and is a standard pytest option that is useful. It stops
in pdb any test that fails.

**Comment by mattclay** [comment]
The option isn't being passed to pytest. Perhaps file(s) are missing from the PR?

**Comment by jmikedupont2** [comment]
I will check when I get home. i think all options are passed through.

File: test/lib/ansible_test/_internal/config.py (hunk 1 above)
Comment (jmikedupont2) [reply test/lib/ansible_test/_internal/config.py:301]: Thank you for catching my silly mistake. I have passed this in.
Tested like this  `strace -f -c -s 9999 python3  bin/ansible-test units --pdb` and `strace -f -c -s 9999 python3  bin/ansible-test units ` and confirmed it was passed through.

**Comment by jmikedupont2** [comment]
Thanks for the code review, this should be better now.

File: test/lib/ansible_test/_internal/cli.py [hunk 2]
```diff
@@ -414,6 +414,13 @@ def parse_args():
                        choices=SUPPORTED_PYTHON_VERSIONS + ('default',),
                        help='python version: %s' % ', '.join(SUPPORTED_PYTHON_VERSIONS))
 
+    units.add_argument('--pdb',
+                       action='store_const',
```
Comment (mattclay) [inline test/lib/ansible_test/_internal/cli.py:418]:
```suggestion
                       action='store_true',
```

File: test/lib/ansible_test/_internal/cli.py [hunk 3]
```diff
@@ -414,6 +414,13 @@ def parse_args():
...
+                       default='unknown',
```
Comment (mattclay) [inline test/lib/ansible_test/_internal/cli.py:419]:
```suggestion
```

File: test/lib/ansible_test/_internal/cli.py [hunk 4]
```diff
@@ -414,6 +414,13 @@ def parse_args():
...
+                       const='yes',
```
Comment (mattclay) [inline test/lib/ansible_test/_internal/cli.py:420]:
```suggestion
```

File: test/lib/ansible_test/_internal/cli.py [hunk 5]
```diff
@@ -414,6 +414,13 @@ def parse_args():
...
+                       dest='pdb',
```
Comment (mattclay) [inline test/lib/ansible_test/_internal/cli.py:421]:
```suggestion
```

File: test/lib/ansible_test/_internal/cli.py [hunk 6]
```diff
@@ -414,6 +414,13 @@ def parse_args():
...
+                       help='pdb on exit')
```
Comment (mattclay) [inline test/lib/ansible_test/_internal/cli.py:422]:
```suggestion
                       help='invoke python debugger on failure')
```

File: test/lib/ansible_test/_internal/config.py (hunk 1 above)
Comment (mattclay) [inline test/lib/ansible_test/_internal/config.py:301]:
```suggestion
        self.pdb = args.pdb  # type: bool
```

**Review by mattclay — CHANGES_REQUESTED** [review]
Please add a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment) describing this change.

Also, an additional change is needed here:

https://github.com/ansible/ansible/blob/a7e4479d018018a6ec288d015cdf64b9a752dac9/test/lib/ansible_test/_internal/delegation.py#L261

That line needs an extra condition added to the end:

```python
or (isinstance(args, UnitsConfig) and args.pdb)
```

**Comment by jmikedupont2** [comment]
Thanks will do this weekend

**Comment by bmillemathias** [comment]
Hi @jmikedupont2
Are you still interested in updating the patch ?
Best.

**Comment by jmikedupont2** [comment]
> Hi @jmikedupont2
> Are you still interested in updating the patch ?
> Best.

yes, I have been OTG camping for the past week, just got this message, will get back to you.

**Pull request closed without merging**